Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,676 @@
+a collection of continuous functional calculi for elements satisfying a given predicate
+`p : A → Prop`, and we require that this predicate is preserved by the functional calculus.
+
+Although `cfcSpec : p a → C(spectrum R a, R) →*ₐ[R] A` is a necessity for getting the full power
+out of the continuous functional calculus, this declaration will generally not be accessed directly
+by the user. One reason for this is that `cfcSpec` requires a proof of `p a` (indeed, if the
+spectrum is empty, there cannot exist a star algebra homomorphism like this). Instead, we provide
+the completely unbundled `cfc : A → (R → R) → A` which operates on bare functions and provides junk
+values when either `a` does not satisfy the property `p`, or else when the function which is the
+argument to `cfc` is not continuous on the spectrum of `a`.
+
+This completely unbundled approach may give up some conveniences, but it allows for tremendous
+freedom. In particular, `cfc a f` makes sense for *any* `a : A` and `f : R → R`. This is quite
+useful in a variety of settings, but perhaps the most important is the following.
+Besides being a star algebra homomorphism sending the identity to `a`, the key property enjoyed
+by the continuous functional calculus is the *composition property*, which guarantees that
+`cfc a (g ∘ f) = cfc (cfc a f) g` under suitable hypotheses on `a`, `f` and `g`. Note that this
+theorem is nearly impossible to state nicely in terms of `cfcSpec` (see `cfcSpec_comp`). An
+additional advantage of the unbundled approach is that expressions like `fun x : R ↦ x⁻¹` are valid
+arguments to `cfc a`, and a bundled continuous counterpart can only make sense when the spectrum of
+`a` does not contain zero and when we have an `⁻¹` operation on the domain.
+
+A reader familiar with C⋆-algebra theory may be somewhat surprised at the level of abstraction here.
+For instance, why not require `A` to be an actual C⋆-algebra? Why define separate continuous
+functional calculi for `R := ℂ`, `ℝ` or `ℝ≥0` instead of simply using the continuous functional
+calculus for normal elements? The reason for both can be explained with a simple example,
+`A := Matrix n n ℝ`. In Mathlib, matrices are not equipped with a norm (nor even a metric), and so
+requiring `A` to be a C⋆-algebra is far too stringent. Likewise, `A` is not a `ℂ`-algebra, and so
+it is impossible to consider the `ℂ`-spectrum of `a : Matrix n n ℝ`.
+
+There is another, more practical reason to define separate continuous functional calculi for
+different scalar rings. It gives us the ability to use functions defined on these types, and the
+algebra of functions on them. For example, in `ℝ` it is quite natural to consider the functions
+`(·⁺ : ℝ → ℝ)` and `(·⁻ : ℝ → ℝ)` because the collection of such functions forms a lattice ordered
+group. If `a : A` is selfadjoint, and we define `a⁺ := cfc a (·⁺ : ℝ → ℝ)`, and likewise for `a⁻`,
```

Comment:
It is not clear to me what you're referring to by "the collection of such functions". Do you mean the "algebra of functions on them"? because here you're mentioning `(·⁺ : ℝ → ℝ)`, not `(·⁺ : (ℝ → ℝ) → ℝ → ℝ)`. Worth rephrasing.